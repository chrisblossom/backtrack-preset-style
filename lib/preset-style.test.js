'use strict';

const transformConfig = require('@backtrack/core/dist/options-file/transform-config')
    .transformConfig;

test('preset returns expected config', () => {
    const backtrackConfig = {
        presets: ['../'],
    };

    const result = transformConfig(backtrackConfig, __dirname);

    expect(result).toMatchSnapshot();
});
