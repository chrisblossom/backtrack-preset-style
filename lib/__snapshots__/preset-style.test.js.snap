// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`option: isApp = true 1`] = `
Object {
  "config": Array [
    Object {},
  ],
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".eslintignore": "<PROJECT_ROOT>/.eslintignore",
        ".eslintrc.js": "<PROJECT_ROOT>/.eslintrc.js",
        ".huskyrc.js": "<PROJECT_ROOT>/.huskyrc.js",
        ".lintstagedrc.js": "<PROJECT_ROOT>/.lintstagedrc.js",
        ".prettierignore": "<PROJECT_ROOT>/.prettierignore",
        ".prettierrc.js": "<PROJECT_ROOT>/.prettierrc.js",
      },
      "allowChanges": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
      "files": Array [
        ".huskyrc.js",
        ".eslintignore",
        ".prettierignore",
        ".lintstagedrc.js",
        ".eslintrc.js",
        ".prettierrc.js",
      ],
      "ignoreUpdates": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
    },
    "makeDirs": Array [
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/eslint.js": "<PROJECT_ROOT>/lib/files/eslint.js",
        "lib/files/ignore": "<PROJECT_ROOT>/lib/files/ignore",
        "lib/files/lint-staged.js": "<PROJECT_ROOT>/lib/files/lint-staged.js",
        "lib/files/prettier.js": "<PROJECT_ROOT>/lib/files/prettier.js",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
      },
      "files": Array [
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
        "lib/files/ignore",
        "lib/files/ignore",
        "lib/files/lint-staged.js",
        "lib/files/eslint.js",
        "lib/files/prettier.js",
      ],
      "hash": Object {
        "lib/files/eslint.js": "3aa22828e051343c67e77b871d094ad536208e8cce15fc270abf97444c68fef9",
        "lib/files/ignore": "df678e7a458aba93c115564cfd4a701bdf4397e943de8745d37f2798c1dc5c11",
        "lib/files/lint-staged.js": "5398a6883c9332018fa32fdac8989989eef1eca27c0e2d92ccf35bb13381e7e7",
        "lib/files/prettier.js": "f76ed92bea30dd6cb22675cedc0044614a9220be9b74859b5750a54ef4745a99",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "d41c387b0d3f16730829f444ef7bc0665cc52e92847c548c5166ebb7597bc69a",
      },
    },
  },
  "format": Array [
    "prettier \\"**/*.{js,mjs,jsx,ts,tsx,json,scss,less,css,md,yml,yaml}\\" --write",
  ],
  "git-pre-commit": Array [
    [Function noop],
    "lint-staged",
  ],
  "git-pre-push": Array [
    [Function noop],
    "backtrack lint",
  ],
  "lint": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx .",
  ],
  "lint.fix": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx --fix .",
  ],
}
`;

exports[`option: node = true 6+ 1`] = `
Object {
  "config": Array [
    Object {},
  ],
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".eslintignore": "<PROJECT_ROOT>/.eslintignore",
        ".eslintrc.js": "<PROJECT_ROOT>/.eslintrc.js",
        ".huskyrc.js": "<PROJECT_ROOT>/.huskyrc.js",
        ".lintstagedrc.js": "<PROJECT_ROOT>/.lintstagedrc.js",
        ".prettierignore": "<PROJECT_ROOT>/.prettierignore",
        ".prettierrc.js": "<PROJECT_ROOT>/.prettierrc.js",
      },
      "allowChanges": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
      "files": Array [
        ".huskyrc.js",
        ".eslintignore",
        ".prettierignore",
        ".lintstagedrc.js",
        ".eslintrc.js",
        ".prettierrc.js",
      ],
      "ignoreUpdates": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
    },
    "makeDirs": Array [
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/eslint-node.js": "<PROJECT_ROOT>/lib/files/eslint-node.js",
        "lib/files/ignore": "<PROJECT_ROOT>/lib/files/ignore",
        "lib/files/lint-staged.js": "<PROJECT_ROOT>/lib/files/lint-staged.js",
        "lib/files/prettier.js": "<PROJECT_ROOT>/lib/files/prettier.js",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
      },
      "files": Array [
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
        "lib/files/ignore",
        "lib/files/ignore",
        "lib/files/lint-staged.js",
        "lib/files/eslint-node.js",
        "lib/files/prettier.js",
      ],
      "hash": Object {
        "lib/files/eslint-node.js": "d29084d1d0cbcc3982525eae881e6e797d1c64be4c6937ab43fb134750c6207d",
        "lib/files/ignore": "df678e7a458aba93c115564cfd4a701bdf4397e943de8745d37f2798c1dc5c11",
        "lib/files/lint-staged.js": "5398a6883c9332018fa32fdac8989989eef1eca27c0e2d92ccf35bb13381e7e7",
        "lib/files/prettier.js": "f76ed92bea30dd6cb22675cedc0044614a9220be9b74859b5750a54ef4745a99",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "d41c387b0d3f16730829f444ef7bc0665cc52e92847c548c5166ebb7597bc69a",
      },
    },
  },
  "format": Array [
    "prettier \\"**/*.{js,mjs,jsx,ts,tsx,json,scss,less,css,md,yml,yaml}\\" --write",
  ],
  "git-pre-commit": Array [
    [Function noop],
    "lint-staged",
  ],
  "git-pre-push": Array [
    [Function noop],
    "backtrack lint",
  ],
  "lint": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx .",
  ],
  "lint.fix": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx --fix .",
  ],
  "prepublishOnly": Array [
    "backtrack lint",
  ],
}
`;

exports[`option: node = true 8+ 1`] = `
Object {
  "config": Array [
    Object {},
  ],
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".eslintignore": "<PROJECT_ROOT>/.eslintignore",
        ".eslintrc.js": "<PROJECT_ROOT>/.eslintrc.js",
        ".huskyrc.js": "<PROJECT_ROOT>/.huskyrc.js",
        ".lintstagedrc.js": "<PROJECT_ROOT>/.lintstagedrc.js",
        ".prettierignore": "<PROJECT_ROOT>/.prettierignore",
        ".prettierrc.js": "<PROJECT_ROOT>/.prettierrc.js",
      },
      "allowChanges": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
      "files": Array [
        ".huskyrc.js",
        ".eslintignore",
        ".prettierignore",
        ".lintstagedrc.js",
        ".eslintrc.js",
        ".prettierrc.js",
      ],
      "ignoreUpdates": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
    },
    "makeDirs": Array [
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/eslint-node.js": "<PROJECT_ROOT>/lib/files/eslint-node.js",
        "lib/files/ignore": "<PROJECT_ROOT>/lib/files/ignore",
        "lib/files/lint-staged.js": "<PROJECT_ROOT>/lib/files/lint-staged.js",
        "lib/files/prettier.js": "<PROJECT_ROOT>/lib/files/prettier.js",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
      },
      "files": Array [
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
        "lib/files/ignore",
        "lib/files/ignore",
        "lib/files/lint-staged.js",
        "lib/files/eslint-node.js",
        "lib/files/prettier.js",
      ],
      "hash": Object {
        "lib/files/eslint-node.js": "d29084d1d0cbcc3982525eae881e6e797d1c64be4c6937ab43fb134750c6207d",
        "lib/files/ignore": "df678e7a458aba93c115564cfd4a701bdf4397e943de8745d37f2798c1dc5c11",
        "lib/files/lint-staged.js": "5398a6883c9332018fa32fdac8989989eef1eca27c0e2d92ccf35bb13381e7e7",
        "lib/files/prettier.js": "f76ed92bea30dd6cb22675cedc0044614a9220be9b74859b5750a54ef4745a99",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "d41c387b0d3f16730829f444ef7bc0665cc52e92847c548c5166ebb7597bc69a",
      },
    },
  },
  "format": Array [
    "prettier \\"**/*.{js,mjs,jsx,ts,tsx,json,scss,less,css,md,yml,yaml}\\" --write",
  ],
  "git-pre-commit": Array [
    [Function noop],
    "lint-staged",
  ],
  "git-pre-push": Array [
    [Function noop],
    "backtrack lint",
  ],
  "lint": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx .",
  ],
  "lint.fix": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx --fix .",
  ],
  "prepublishOnly": Array [
    "backtrack lint",
  ],
}
`;

exports[`preset returns expected config no options 6+ 1`] = `
Object {
  "config": Array [
    Object {},
  ],
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".eslintignore": "<PROJECT_ROOT>/.eslintignore",
        ".eslintrc.js": "<PROJECT_ROOT>/.eslintrc.js",
        ".huskyrc.js": "<PROJECT_ROOT>/.huskyrc.js",
        ".lintstagedrc.js": "<PROJECT_ROOT>/.lintstagedrc.js",
        ".prettierignore": "<PROJECT_ROOT>/.prettierignore",
        ".prettierrc.js": "<PROJECT_ROOT>/.prettierrc.js",
      },
      "allowChanges": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
      "files": Array [
        ".huskyrc.js",
        ".eslintignore",
        ".prettierignore",
        ".lintstagedrc.js",
        ".eslintrc.js",
        ".prettierrc.js",
      ],
      "ignoreUpdates": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
    },
    "makeDirs": Array [
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/eslint.js": "<PROJECT_ROOT>/lib/files/eslint.js",
        "lib/files/ignore": "<PROJECT_ROOT>/lib/files/ignore",
        "lib/files/lint-staged.js": "<PROJECT_ROOT>/lib/files/lint-staged.js",
        "lib/files/prettier.js": "<PROJECT_ROOT>/lib/files/prettier.js",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
      },
      "files": Array [
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
        "lib/files/ignore",
        "lib/files/ignore",
        "lib/files/lint-staged.js",
        "lib/files/eslint.js",
        "lib/files/prettier.js",
      ],
      "hash": Object {
        "lib/files/eslint.js": "3aa22828e051343c67e77b871d094ad536208e8cce15fc270abf97444c68fef9",
        "lib/files/ignore": "df678e7a458aba93c115564cfd4a701bdf4397e943de8745d37f2798c1dc5c11",
        "lib/files/lint-staged.js": "5398a6883c9332018fa32fdac8989989eef1eca27c0e2d92ccf35bb13381e7e7",
        "lib/files/prettier.js": "f76ed92bea30dd6cb22675cedc0044614a9220be9b74859b5750a54ef4745a99",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "d41c387b0d3f16730829f444ef7bc0665cc52e92847c548c5166ebb7597bc69a",
      },
    },
  },
  "format": Array [
    "prettier \\"**/*.{js,mjs,jsx,ts,tsx,json,scss,less,css,md,yml,yaml}\\" --write",
  ],
  "git-pre-commit": Array [
    [Function noop],
    "lint-staged",
  ],
  "git-pre-push": Array [
    [Function noop],
    "backtrack lint",
  ],
  "lint": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx .",
  ],
  "lint.fix": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx --fix .",
  ],
  "prepublishOnly": Array [
    "backtrack lint",
  ],
}
`;

exports[`preset returns expected config no options 8+ 1`] = `
Object {
  "config": Array [
    Object {},
  ],
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".eslintignore": "<PROJECT_ROOT>/.eslintignore",
        ".eslintrc.js": "<PROJECT_ROOT>/.eslintrc.js",
        ".huskyrc.js": "<PROJECT_ROOT>/.huskyrc.js",
        ".lintstagedrc.js": "<PROJECT_ROOT>/.lintstagedrc.js",
        ".prettierignore": "<PROJECT_ROOT>/.prettierignore",
        ".prettierrc.js": "<PROJECT_ROOT>/.prettierrc.js",
      },
      "allowChanges": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
      "files": Array [
        ".huskyrc.js",
        ".eslintignore",
        ".prettierignore",
        ".lintstagedrc.js",
        ".eslintrc.js",
        ".prettierrc.js",
      ],
      "ignoreUpdates": Object {
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".huskyrc.js": false,
        ".lintstagedrc.js": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
      },
    },
    "makeDirs": Array [
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/eslint.js": "<PROJECT_ROOT>/lib/files/eslint.js",
        "lib/files/ignore": "<PROJECT_ROOT>/lib/files/ignore",
        "lib/files/lint-staged.js": "<PROJECT_ROOT>/lib/files/lint-staged.js",
        "lib/files/prettier.js": "<PROJECT_ROOT>/lib/files/prettier.js",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
      },
      "files": Array [
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js",
        "lib/files/ignore",
        "lib/files/ignore",
        "lib/files/lint-staged.js",
        "lib/files/eslint.js",
        "lib/files/prettier.js",
      ],
      "hash": Object {
        "lib/files/eslint.js": "3aa22828e051343c67e77b871d094ad536208e8cce15fc270abf97444c68fef9",
        "lib/files/ignore": "df678e7a458aba93c115564cfd4a701bdf4397e943de8745d37f2798c1dc5c11",
        "lib/files/lint-staged.js": "5398a6883c9332018fa32fdac8989989eef1eca27c0e2d92ccf35bb13381e7e7",
        "lib/files/prettier.js": "f76ed92bea30dd6cb22675cedc0044614a9220be9b74859b5750a54ef4745a99",
        "node_modules/@backtrack/preset-git-hooks/lib/files/husky.js": "d41c387b0d3f16730829f444ef7bc0665cc52e92847c548c5166ebb7597bc69a",
      },
    },
  },
  "format": Array [
    "prettier \\"**/*.{js,mjs,jsx,ts,tsx,json,scss,less,css,md,yml,yaml}\\" --write",
  ],
  "git-pre-commit": Array [
    [Function noop],
    "lint-staged",
  ],
  "git-pre-push": Array [
    [Function noop],
    "backtrack lint",
  ],
  "lint": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx .",
  ],
  "lint.fix": Array [
    "eslint --ext .js,.jsx,.mjs,.ts,.tsx --fix .",
  ],
  "prepublishOnly": Array [
    "backtrack lint",
  ],
}
`;
